lib/src/universal_ble_platform_interface.dart

```diff
@@ -105,23 +153,25 @@ abstract class UniversalBlePlatform {
     if (_pairStateMap[deviceId] == isPaired) return;
     _pairStateMap[deviceId] = isPaired;
 
+    _pairStateStreamController.add((deviceId: deviceId, isPaired: isPaired));
+
     try {
       onPairingStateChange?.call(deviceId, isPaired);
     } catch (_) {}
   }
 
-  /// Creates an auto disposable streamController
-  void _setupConnectionStreamIfRequired() {
-    if (_connectionStreamController != null) return;
-
-    _connectionStreamController = StreamController.broadcast();
-
-    // Auto dispose if no more subscribers
-    _connectionStreamController?.onCancel = () {
-      // logInfo('Disposing Connection Stream');
-      _connectionStreamController?.close();
-      _connectionStreamController = null;
-    };
+  /// Helper function to wait for Connection result from Stream and also handles connection errors
+  StreamSubscription waitConnection(
```

Comment:
The name should be better. Currently it implies that it is a future but it is a stream.

Can this be private? I am reviewing on a phone so it's hard to search where it's being used.

Reply:
Yeah this is suppose to be private, and not exposed as public api

Reply:
Removed this method